### ISSUE TITLE
The no-WebGPU error blames the runtime when the real cause is often no GPU

### ISSUE THREAD

**Issue by felladrin** [felladrin/gguf-trainer#84]
`pretrain` and `finetune` both die with:

```
no WebGPU: training needs Deno (Node and Bun have no GPU backend here)
```

but `initWebGPU()` returns `null` for two different reasons:

```ts
if (!nav?.gpu) return null;                       // the runtime has no WebGPU: Node, Bun
const adapter = await nav.gpu.requestAdapter();
if (!adapter) return null;                        // Deno, but no usable GPU on the machine
```

The message only describes the first. Someone running Deno on a machine with no GPU is told that training needs Deno, which they are already using, and the real cause goes unnamed. That is not hypothetical: it is exactly the situation in #41, a Ryzen 5 PRO 3600 with no discrete GPU.

Distinguishing them costs one boolean out of `initWebGPU`, or a second probe at the call site:

- no `navigator.gpu`: "training needs Deno; Node and Bun have no GPU backend here."
- no adapter: "no GPU adapter found. Training needs one; this machine's CPU is not a fallback, since the CPU backend here is a single-threaded correctness reference." Then point at what to do instead, the way the `--loss-chunk` and OOM messages do.

Worth adding the second half to the readme too. The current text says training needs a WebGPU device, which is true, but nothing says why the CPU path that exists (`trainLM`, used by `demo` and by every parity test) is not offered as an option. Measured on this machine, same shape and steps, CPU `trainLM` against GPU `trainLMGpuResident`:

| shape | CPU | GPU |
| --- | ---: | ---: |
| 6.0M params, seq 256 | 15.8 tok/s | 786.4 tok/s |
| 32.0M params, seq 256 | 1.6 tok/s | |

5.3x the parameters costs 9.9x the time, so it is worse than linear, and a linear extrapolation to a 596M model is already 0.086 tok/s: about 12 seconds per token, or years for a fine-tune worth running. That is the honest reason there is no `--cpu` for training, and the error message is the place a user learns it.
